packages/mobile/App/ui/components/MultiSelectModal/MultiSelectModalField.tsx

```diff
@@ -41,35 +45,42 @@ export const MultiSelectModalField = ({
   const navigation = useNavigation();
   const [label, setLabel] = useState(null);
 
-  const appendLabel = (items: OptionType[]) => {
-    return items.map(x => ` ${x.label}`).toString();
-  };
-  const onPress = (selectedItems: OptionType[]): void => {
+  const handleSaveCallback = (selectedItems: OptionType[]): void => {
     onChange(selectedItems);
-    setLabel(appendLabel(selectedItems));
+    setLabel(extractLabel(selectedItems));
   };
 
   const openModal = (): void =>
     navigation.navigate(modalRoute, {
-      callback: onPress,
+      callback: handleSaveCallback,
       suggester,
       modalTitle,
       value,
       searchPlaceholder,
     });
-  const loadDefaultValues = async (values: string[]) => {
-    const _values: OptionType[] = [];
-    values.forEach(async x => {
+
+  // This function is not in use, however, it's meant to be used when
+  // initial values are set. Pay extra care to the value format it expects.
+  const loadInitialLabel = useCallback(async (values: string[]) => {
+    if (values.length === 0) {
+      return;
+    }
+    const selectedValues: OptionType[] = [];
+    for (const x of values) {
       const data = await suggester.fetchCurrentOption(x);
-      _values.push(data);
-    });
-    const _label = appendLabel(_values);
-    setLabel(_values.length > 0 ? _label : setLabel(placeholder));
-  };
+      selectedValues.push(data);
+    }
+
+    const updatedLabel = extractLabel(selectedValues);
+    setLabel(updatedLabel);
+  }, [suggester]);
 
   useEffect(() => {
-    if (!suggester || !value) return;
-    loadDefaultValues(value);
+    if (!value) return;
+    loadInitialLabel(value);
+  // Disabling the linter because we only want to run this on mount
+  // if the component has an initial value and ignore value updates.
+  // eslint-disable-next-line react-hooks/exhaustive-deps
```

Comment:
thanks for the comment - it's quite an annoying lint rule as running on mount is one of the main use cases for useEffect!